core/services/RestService.js

```diff
@@ -223,7 +234,13 @@ module.exports = {
       const options = {
         populate: ctx.query.populate && ctx.query.populate.split(","),
       }
-      ctx.body = await yaxys.db.insert(identity, ctx.request.body, options)
+
+      try {
+        ctx.body = await yaxys.db.knex.transaction((trx) =>
+          yaxys.db.insert(identity, ctx.request.body, options, trx))
+      } catch (err) {
+        ctx.throw(500)
+      }
     }
   },
 }
```

Comment:
If you do it that way, remove 3 transaction-related methods from the Adapter – and tests for them